client/lib/analytics/utils.js

```diff
@@ -95,3 +101,17 @@ export function shouldSkipAds() {
 	debug( `Is Skipping Ads: ${ result }` );
 	return result;
 }
+
+/**
+ * Check if a path should report the currently selected site ID.
+ *
+ * Some paths should never report it because it's used
+ * to tell general admin and site-specific activities apart.
+ *
+ * @param {String} path The tracked path.
+ * @returns {Boolean} If the report should null `blog_id`.
+ */
+export const shouldReportOmitBlogId = path =>
+	'/' === path ||
+	'/me' === path ||
```

Comment:
Ah so we can still run into the same issue if we add short routes to this list in the future. Let's try for a more general solution.